lib/broadway/message.ex

```diff
@@ -18,13 +18,15 @@ defmodule Broadway.Message do
           data: any,
           acknowledger: {module, data :: any},
           publisher: atom,
-          processor_pid: pid
+          processor_pid: pid,
+          status: :pending | :processed | {:failed, reason :: any}
```

Comment:
I would like to avoid the status `:processed` because if we have multiple processors in the future, when we would actually set it? I would go with `status: :ok | {:failed, String.t}` for now. Thoughts?